memory_statistics/memory_statistics.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+
+import os
+import argparse
+import subprocess
+from collections import defaultdict
+
+
+__TABLE_HEADER__ = "Code Size of {} (example generated with GCC for ARM Cortex-M)"
+__THIS_FILE_PATH__ = os.path.dirname(os.path.abspath(__file__))
+
+
+def make(sources, includes, flags, opt):
+    args = ['make', '-B', '-f', __THIS_FILE_PATH__ + '/makefile']
+
+    cflags = f'-O{opt} -I "{__THIS_FILE_PATH__ + "/config_files"}"'
+    for inc_dir in includes.splitlines():
+        cflags += ' -I "'+os.path.abspath(inc_dir)+'"'
+    for flag in flags.splitlines():
+        cflags += f' -D {flag}'
+
+    args += ['CFLAGS=' + cflags]
+    args += ['SRCS=' + " ".join(sources.splitlines())]
+
+    proc = subprocess.Popen(args, stdout=subprocess.PIPE, universal_newlines=True)
+    (results, _) = proc.communicate()
+    print(results)
+
+    return results
+
+
+def convert_size_to_kb(byte_size):
+    kb_size = round(int(byte_size)/1024,1)
+    if (kb_size == 0.0):
+        return 0.1
+    else:
+        return kb_size
+
+
+def parse_make_output(output, values, key):
+    '''
+    It assumes the Berkley output format:
+    text       data     bss     dec     hex filename
+
+    Values is defaultdict mapping filenames to dicts
+    This adds each file's size to the dict in its entry under the provided key.
+    '''
+    output = output.splitlines()
+
+    # Skip to size output
+    while output:
+        line = output[0]
+        output = output[1:]
+        if line.startswith('arm-none-eabi-size'):
+            break
+    # Skip header
+    output = output[1:]
+
+    for line in output:
+        parts = line.split()
+
+        filename = os.path.basename(parts[5])
```

Comment:
It will be good to mention the expected format and what are part[5] and part[0] in comments.